src/assets/js/themes.js

```diff
@@ -1,49 +1,48 @@
 /* theme toggle buttons */
-(function() {
-    var enableToggle = function(btn) {
+(function () {
+    var enableToggle = function (btn) {
         btn.setAttribute("aria-pressed", "true");
     }
-
-    var disableToggle = function(btn) {
+    var disableToggle = function (btn) {
         btn.setAttribute("aria-pressed", "false");
     }
+    var setTheme = function (theme) {
+        document.documentElement.setAttribute('data-theme', theme);
+        window.localStorage.setItem("theme", theme);
+    }
 
+    var theme = window.localStorage.getItem("theme");
+    if (!theme || (theme !== "light" && theme !== "dark")) {
+        var isDark = window.matchMedia('(prefers-color-scheme: dark)').matches;
+        theme = isDark ? "dark" : "light";
+    }
 
-    let theme = window.localStorage.getItem("theme");
-
-    document.addEventListener('DOMContentLoaded', function() {
+    document.addEventListener('DOMContentLoaded', function () {
         var switcher = document.getElementById('js-theme-switcher');
         switcher.removeAttribute('hidden');
-
         var light_theme_toggle = document.getElementById('light-theme-toggle'),
             dark_theme_toggle = document.getElementById('dark-theme-toggle');
-
-
-        if (theme == "light") {
+        if (theme === "light") {
             enableToggle(light_theme_toggle);
             disableToggle(dark_theme_toggle);
-        } else if (theme == "dark") {
+            setTheme(theme);
```

Comment:
I think we don't need this part of the code too... 
- https://github.com/eslint/new.eslint.org/blob/7bd0f8f4c3be64337f2392203cf952d8a512143b/src/_includes/layouts/base.html#L68-L73

During initial page load this method actually setstheme as dark mode if the system preference is dark else we dont set any theme.

Reply:
Yes, the above code works perfectly if the user preference is dark but if not then we are not setting any theme. 

https://github.com/eslint/new.eslint.org/blob/7bd0f8f4c3be64337f2392203cf952d8a512143b/src/assets/scss/animation.scss#L104-L114

In styles using theme only, we are deciding which dropdown (light or dark) should be shown. So the above block of code is not working as expected.

One way we can set the theme to light is if the system preference is not dark.
```
    if (!theme && window.matchMedia('(prefers-color-scheme: dark)').matches) { 
	         document.documentElement.setAttribute('data-theme', 'dark'); 
	} else
	  document.documentElement.setAttribute('data-theme', 'light'); //like this
```
But if we do like this it will break the theme toggle logic. It will always be based on the system's preference.

So in the theme.js I am initially setting the theme based on preference and updating it in the local store. 
If the theme is set once the below logic will set the theme attribute based on the local store data.
https://github.com/eslint/new.eslint.org/blob/7bd0f8f4c3be64337f2392203cf952d8a512143b/src/assets/scss/animation.scss#L104-L114

And manual toggling using the toggle button in UI will be handled by the below code.

https://github.com/eslint/new.eslint.org/blob/cc53724d80a2200a724e3756c38e7e06b1482fc1/src/assets/js/themes.js#L34-L46


Correct me if I am wrong and open to suggestions/comments.

Reply:
Yea I get it :) I just thought it was better to have the initial theme setting done in the base.html instead of doing similar stuff at `base.html` as well as `theme.js`. But anything is fine with me too :)